src/game/client/ingameui.cpp

```diff
@@ -1751,3 +1765,1231 @@ void InGameUI::Recreate_Control_Bar()
+}
+
+CommandButton const *InGameUI::Get_GUI_Command()
+{
+    return m_pendingGUICommand;
+}
+
+ThingTemplate const *InGameUI::Get_Pending_Place_Type()
+{
+    return m_pendingPlaceType;
+}
+
+ObjectID const InGameUI::Get_Pending_Place_Source_Object_ID()
+{
+    return m_pendingPlaceSourceObjectID;
+}
+
+#ifndef GAME_DEBUG_STRUCTS
+bool InGameUI::Get_Prevent_Left_Click_Deselection_In_Alternate_Mouse_Mode_For_One_Click()
+{
+    return m_preventLeftClickDeselectionInAlternateMouseModeForOneClick;
+}
+
+void InGameUI::Set_Prevent_Left_Click_Deselection_In_Alternate_Mouse_Mode_For_One_Click(bool prevent)
+{
+    m_preventLeftClickDeselectionInAlternateMouseModeForOneClick = prevent;
+}
+#endif
+
+void InGameUI::Set_Placement_Start(ICoord2D const *start)
+{
+    if (start != nullptr) {
+        m_placeAnchorStart.x = start->x;
+        m_placeAnchorStart.y = start->y;
+        m_placeAnchorEnd.x = start->x;
+        m_placeAnchorEnd.y = start->y;
+        m_placeAnchorInProgress = true;
+    } else {
+        m_placeAnchorInProgress = false;
+    }
+}
+
+void InGameUI::Set_Placement_End(ICoord2D const *end)
+{
+    if (end != nullptr) {
+        m_placeAnchorEnd.x = end->x;
+        m_placeAnchorEnd.y = end->y;
+    }
+}
+
+bool InGameUI::Is_Placement_Anchored()
+{
+    return m_placeAnchorInProgress;
+}
+
+void InGameUI::Get_Placement_Points(ICoord2D *start, ICoord2D *end)
+{
+    if (start != nullptr) {
+        *start = m_placeAnchorStart;
+    }
+
+    if (end != nullptr) {
+        *end = m_placeAnchorEnd;
+    }
+}
+
+float InGameUI::Get_Placement_Angle()
+{
+    if (m_placeIcon[0] != nullptr) {
+        return m_placeIcon[0]->Get_Orientation();
+    } else {
+        return 0.0f;
+    }
+}
+
+void InGameUI::Select_Drawable(Drawable *drawable)
+{
+    if (!drawable->Is_Selected()) {
+        m_frameSelectionChanged = g_theGameLogic->Get_Frame();
+        drawable->Friend_Set_Selected();
+        m_selectedDrawables.push_front(drawable);
+        Increment_Select_Count();
+        Evaluate_Solo_Nexus(drawable);
+        g_theControlBar->On_Drawable_Selected(drawable);
+    }
+}
+
+void InGameUI::Evaluate_Solo_Nexus(Drawable *drawable)
+{
+    m_soloNexusSelectedDrawableID = INVALID_DRAWABLE_ID;
+    Object *object;
+
+    if (drawable == nullptr || (object = drawable->Get_Object()) == nullptr || object->Is_KindOf(KINDOF_MOB_NEXUS)
+        || object->Is_KindOf(KINDOF_IGNORED_IN_GUI)) {
+        short count = 0;
+
+        for (auto it = m_selectedDrawables.begin(); it != m_selectedDrawables.end(); it++) {
+            Drawable *draw = *it;
+            Object *obj = draw->Get_Object();
+
+            if (obj != nullptr) {
+                if (obj->Is_KindOf(KINDOF_MOB_NEXUS)) {
+                    if (++count != 1) {
+                        m_soloNexusSelectedDrawableID = INVALID_DRAWABLE_ID;
+                        return;
+                    }
+
+                    m_soloNexusSelectedDrawableID = drawable->Get_ID();

```

Comment:
This does not look right.

Above there is
```c++
if (drawable == nullptr || ...)
```
And here it can dereference `drawable` when it is null.

Reply:
fixed.